crates/bifrost/src/providers/kafka_loglet/mod.rs

```diff
@@ -239,7 +250,13 @@ impl LogletBase for KafkaLoglet {
     }
 
     async fn find_tail(&self) -> crate::Result<TailState<Self::Offset>, OperationError> {
-        Ok(TailState::Open(Self::Offset::OLDEST))
+        let consumer: BaseConsumer =
+            initialize_consumer(&self.client_config, &self.topic, self.partition)?;
```

Comment:
Doesn't really matter but we could also make the consumer a field of the `KafkaLoglet` struct.